erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -870,36 +870,45 @@ frappe.ui.form.on('Payment Entry', {
 		var total_deductions = frappe.utils.sum($.map(frm.doc.deductions || [],
 			function(d) { return flt(d.amount) }));
 
-		if(frm.doc.party) {
-			if(frm.doc.payment_type == "Receive"
-				&& frm.doc.base_total_allocated_amount < frm.doc.base_received_amount + total_deductions
-				&& frm.doc.total_allocated_amount < frm.doc.paid_amount + (total_deductions / frm.doc.source_exchange_rate)) {
-					unallocated_amount = (frm.doc.base_received_amount + total_deductions + frm.doc.base_total_taxes_and_charges
-						- frm.doc.base_total_allocated_amount) / frm.doc.source_exchange_rate;
-			} else if (frm.doc.payment_type == "Pay"
-				&& frm.doc.base_total_allocated_amount < frm.doc.base_paid_amount - total_deductions
-				&& frm.doc.total_allocated_amount < frm.doc.received_amount + (total_deductions / frm.doc.target_exchange_rate)) {
-					unallocated_amount = (frm.doc.base_paid_amount + frm.doc.base_total_taxes_and_charges - (total_deductions
-						+ frm.doc.base_total_allocated_amount)) / frm.doc.target_exchange_rate;
-			}
+		const {
+			party,
+			payment_type,
+			base_paid_amount,
+			base_received_amount,
+			base_total_taxes_and_charges,
+			total_allocated_amount,
+			target_exchange_rate,
+			source_exchange_rate,
+		} = frm.doc;
+
+		const exchange_rate = payment_type === "Pay" ? target_exchange_rate : source_exchange_rate;
+		const base_amount = payment_type === "Pay" ? base_paid_amount : base_received_amount;
+
+		if (party &&
+			total_allocated_amount < (base_amount - total_deductions) / exchange_rate) {
+			unallocated_amount = (
```

Comment:
@dj12djdjs I am not sure if the deduction still has any impact on the calculation of `unallocated_amount`. If you'll check the formula the net total deduction is still 0 as the same value gets added and subtracted. Will be helpful if you add an example/test case here with a deduction in the case of Receive and Pay as well.

Also this unit test `test_multiple_payment_entries_against_expense ` seems to be related to this change. This will also have to be fixed